lib/models/release.coffee

```diff
@@ -180,6 +188,162 @@ getReleaseModel = (deps, opts) ->
 					, options
 		.asCallback(callback)
 
+	###*
+	# @namespace resin.models.release.tags
+	# @memberof resin.models.release
+	###
+	exports.tags = {}
+
+	###*
+	# @summary Get all release tags for an application
+	# @name getAllByApplication
+	# @public
+	# @function
+	# @memberof resin.models.release.tags
+	#
+	# @param {String|Number} nameOrId - application name (string) or id (number)
+	# @param {Object} [options={}] - extra pine options to use
+	# @fulfil {Object[]} - release tags
+	# @returns {Promise}
+	#
+	# @example
+	# resin.models.release.tags.getAllByApplication('MyApp').then(function(tags) {
+	# 	console.log(tags);
+	# });
+	#
+	# @example
+	# resin.models.release.tags.getAllByApplication(999999).then(function(tags) {
+	# 	console.log(tags);
+	# });
+	#
+	# @example
+	# resin.models.release.tags.getAllByApplication('MyApp', function(error, tags) {
+	# 	if (error) throw error;
+	# 	console.log(tags)
+	# });
+	###
+	exports.tags.getAllByApplication = (nameOrId, options = {}, callback) ->
+		applicationModel().get(nameOrId, $select: 'id').get('id').then (id) ->
+			tagsModel().getAll(
+				mergePineOptions
+					$filter:
+						release:
+							$any:
+								$alias: 'r',
+								$expr: r: belongs_to__application: id
+				, options
+			)
+		.asCallback(callback)
+
+	###*
+	# @summary Get all release tags for a release
+	# @name getAllByRelease
+	# @public
+	# @function
+	# @memberof resin.models.release.tags
+	#
+	# @param {Number} id - release id
+	# @param {Object} [options={}] - extra pine options to use
+	# @fulfil {Object[]} - release tags
+	# @returns {Promise}
+	#
+	# @example
+	# resin.models.release.tags.getAllByRelease(123).then(function(tags) {
+	# 	console.log(tags);
+	# });
+	#
+	# @example
+	# resin.models.release.tags.getAllByRelease(123, function(error, tags) {
+	# 	if (error) throw error;
+	# 	console.log(tags)
+	# });
+	###
+	exports.tags.getAllByRelease = (id, options = {}, callback) ->
+		callback = findCallback(arguments)
+
+		Promise.all([
+			# only doing this so that the user get a ResinReleaseNotFound
+			# error when the release doean't actually exist
+			exports.get(id, $select: 'id').get('id')
+			# start the request on parallel, since we don't need
+			# any extra data to wait for
+			exports.tags.getAll(
+				mergePineOptions
+					$filter: release: id
+				, options
+			)
```

Comment:
We could do these two together, couldn't we? Can't you get the release by id, and expand to get the tags too? I think that will throw properly if the release doesn't exist, and let you grab the tags from there very easily as well.